File: problems/Minimum Number of Arrows to Burst Balloons/findMinArrowShots.cs
public class Solution {
    public int FindMinArrowShots(int[][] points) {
        if (points.Length == 0)
        {
            return 0;
        }

        Array.Sort(points, (i1, i2) =>
        {
            var cmpEnd = i1[0].CompareTo(i2[0]);
            if (cmpEnd != 0)
            {
                return cmpEnd;
            }
            return i1[1].CompareTo(i2[1]);
        });

        int i = 0;
        int j = 1;
        int res = 0;

        while (i < points.Length)
        {
            int to = points[i][1];

            while (j < points.Length && points[j][0] <= to)
            {
                to = Math.Min(to, points[j][1]);
                j++;
            }

            res++;

            i = j;
            j = i + 1;
        }

        return res;
    }
}

File: problems/Minimum Number of Arrows to Burst Balloons/findMinArrowShots.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        int result = 0;
        int n = points.length;
        int i = 0;
        int j = 1;

        if (0 == n) {
            return 0;
        }

        Arrays.sort(points, (a,b)->
        {
            if (0 == Integer.compare(a[0], b[0])) {
                return Integer.compare(a[1],b[1]);
            } else {
                return Integer.compare(a[0],b[0]);
            }
        });

        while (n > i) {
            int to = points[i][1];

            while (n > j && points[j][0] <= to) {
                to = Math.min(to, points[j][1]);
                ++j;
            }

            ++result;
            i = j;
            j = i + 1;
        }

        return result;
    }
}
